runtime/compiler/control/JitDump.cpp

```diff
@@ -483,9 +483,98 @@ runJitdump(char *label, J9RASdumpContext *context, J9RASdumpAgent *agent)
+         // signature and get back a J9Method* if it exists? Right? I've searched but I cannot find this API. I
+         // will tag this in a review and hopefully someone knows the right answer here.
+         J9Method* userMethod = NULL;
+
+            {
+            TR::VMAccessCriticalSection findUserMethod(TR_J9VMBase::get(jitConfig, crashedThread));
+
+            char* indexOfDot = strchr(agent->dumpOptions, '.');
+            if (indexOfDot != NULL)
+               {
+               J9Class* userClass = crashedThread->javaVM->internalVMFunctions->internalFindClassUTF8(crashedThread,
+                  reinterpret_cast<U_8*>(agent->dumpOptions), indexOfDot - agent->dumpOptions, crashedThread->javaVM->systemClassLoader, J9_FINDCLASS_FLAG_EXISTING_ONLY);
+
+               if (userClass != NULL)
+                  {
+                  J9JNINameAndSignature nameAndSig;
+                  nameAndSig.name = indexOfDot + 1;
+
+                  char* indexOfParen = strchr(agent->dumpOptions, '(');
+                  if (indexOfParen != NULL)
+                     {
+                     nameAndSig.nameLength = indexOfParen - nameAndSig.name;
+                     nameAndSig.signature = indexOfParen;
+                     nameAndSig.signatureLength = strlen(nameAndSig.signature);
+
+                     userMethod = reinterpret_cast<J9Method*>(crashedThread->javaVM->internalVMFunctions->javaLookupMethod(crashedThread,
+                        userClass, reinterpret_cast<J9ROMNameAndSignature*>(&nameAndSig), NULL, J9_LOOK_JNI | J9_LOOK_NO_JAVA));
+                     }
+                  }
+               }
+            }
+
+         if (userMethod != NULL)
+            {
+            // TODO: Need to check with reviewers whether this will always exist if a method has been compiled before.
+            // Also should think about whether it is useful to have this restriction? Would there ever be a need to
+            // compile a method that has never been compiled before? Off top my head I would think not, but perhaps
+            // someone else may think of a useful scenario.
+            TR_PersistentJittedBodyInfo* bodyInfo = TR::Recompilation::getJittedBodyInfoFromPC(userMethod->extra);
+            if (bodyInfo != NULL)
+               {
+               recompilationThreadInfo->resumeCompilationThread();
+
+               if (threadCompInfo == NULL)
+                  {
+                  // We are an application thread
+                  TR_CompilationErrorCode compErrCode = recompileMethodForLog(
+                     crashedThread,
+                     userMethod,
+                     compInfo,
+                     bodyInfo->getHotness(),
+                     bodyInfo->getIsProfilingBody(),
+                     NULL,
+                     bodyInfo->getIsAotedBody(),
+                     bodyInfo->getStartPCAfterPreviousCompile(),
+                     logFile
+                  );
+                  }
```

Comment:
@dsouzai / @mpirvu looking for your thoughts on this comment here. Does the body info always exist if there was a previous compilation? Do we envision any use for compiling methods which have never been compiled before here?

Commenting on this again because I had to force push to resolve a conflict and the review was then left in "Outdated" state on GitHub so I was worried you may not see this.

Reply:
It looks like we'll always create the bodyinfo as long as recompilation is supported on the platform. However, the bodyinfo may not always be in the preprologue. I think the better thing to do here would be to do

```
J9JITExceptionTable metaData = NULL;
TR_PersistentJittedBodyInfo* bodyInfo = NULL;

{
TR_TranslationArtifactManager *artifactManager = TR_TranslationArtifactManager::getGlobalArtifactManager();
TR_TranslationArtifactManager::CriticalSection jitdumpFindBodyInfo;

metaData 
   = reinterpret_cast<J9JITExceptionTable *>(
      artifactManager->retrieveArtifact(
         reinterpret_cast<uintptr_t>(userMethod->extra)));
}

if (metadata & metadata->bodyInfo)
   bodyInfo = metadata->bodyInfo;
```
That way the only way you're going to get a NULL bodyInfo is if recomp isn't supported on that platform.

Reply:
Thanks. Even better we can use the VM APIs which are readily available. Fixed in 12f8b75.